jbmc/src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -124,7 +125,32 @@ void java_bytecode_languaget::get_language_options(const cmdlinet &cmd)
     std::back_inserter(extra_methods),
     build_load_method_by_regex);
 
-  if(cmd.isset("java-cp-include-files"))
+  if(cmd.isset("load-containing-class-only"))
+  {
+    const std::string function_to_analyse = cmd.get_value("function");
+    const std::size_t first_dollar_index =
+      function_to_analyse.find_first_of('$');
+    const std::size_t last_dot_index = function_to_analyse.find_last_of('.');
+    std::string class_name = function_to_analyse;
+    if(first_dollar_index != std::string::npos)
+      class_name = class_name.substr(0, first_dollar_index);
+    else
+      class_name = class_name.substr(0, last_dot_index);
+    replace_all(class_name, ".", "\\/");
+    java_cp_include_files = class_name +
+                            "(\\$.*)?\\.class|"
+                            "java\\/.*|"
+                            "javax\\/.*|"
+                            "sun\\/.*|"
+                            "org\\/cprover\\/.*|"
+                            "com\\/diffblue\\/annotation\\/.*";
```

Comment:
This is obviously not great. Please suggest ways of making this more general.

Reply:
Not actually making it more general, but you might do `(javax|sun|org\\/cprover|com\\/diffblue\\/annotation)\\/.*`

Reply:
These hard-coded values make the option a bit misleading (it is not only loading the containing class). Maybe these could be added through java-cp-include-files as additional files to include (that have to be given explicitly) on the command line.

Reply:
@romainbrenguier That might indeed be better. Will try that.